reasoner/resolution/resolver/RetrievableResolver.java

```diff
@@ -35,19 +39,25 @@
     protected final Map<ConceptMap, Driver<BoundRetrievableResolver>> boundResolvers;
 
     public RetrievableResolver(Driver<RetrievableResolver> driver, Retrievable retrievable, ResolverRegistry registry) {
-        super(driver, RetrievableResolver.class.getSimpleName() + "(pattern: " + retrievable.pattern() + ")", registry);
+        super(driver, RetrievableResolver.class.getSimpleName() + "(pattern: " + retrievable.pattern() + ")",
+              createEquivalentMappings(retrievable), registry);
         this.retrievable = retrievable;
         this.boundResolvers = new HashMap<>();
     }
 
     @Override
-    Driver<BoundRetrievableResolver> getOrCreateBoundResolver(AnswerState.Partial<?> partial) {
-        return boundResolvers.computeIfAbsent(partial.conceptMap(), p -> {
-            LOG.debug("{}: Creating a new BoundRetrievableResolver for bounds: {}", name(), partial);
-            return registry.registerBoundRetrievable(retrievable, partial.conceptMap());
+    protected Driver<BoundRetrievableResolver> getOrCreateBoundResolver(AnswerState.Partial<?> partial, ConceptMap mapped) {
```

Comment:
we should pass the mapping in, which couples the two arguments together. The instinct is that the `mapped` you pass in should be called `bounds`, but then it's unclear whether to use the bounds from the partial or the mapped. Also, `mapped` is domain knowledge from above ie. outside this function. We fix all those by passing in `mapping`